xml/vt_qemu_ga.xml

```diff
@@ -0,0 +1,174 @@
+
+  <para>
+   For example:
+  </para>
+
+<screen>&prompt.sudo;virsh qemu-agent-command sle15sp2 '{"execute":"guest-info"}' --pretty
+{
+"return": {
+  "version": "4.2.0",
+  "supported_commands": [
+    {
+      "enabled": true,
+      "name": "guest-get-osinfo",
+      "success-response": true
+    },
+[...]
+</screen>
+ </sect1>
+ <sect1 xml:id="cha-qemu-ga-commands">
+  <title>&virsh; Commands that Require &qemu; GA</title>
+
+  <para>
+   Several &virsh; commands require &qemu; GA for their functionality. Here are
+   some of them:
+  </para>
+
+  <variablelist>
+   <varlistentry>
+    <term><command>virsh guestinfo</command></term>
+    <listitem>
+     <para>
+      Prints information about the guest from the guest's point of view.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><command>virsh guestvcpus</command></term>
+    <listitem>
+     <para>
+      Querys or change state of virtual CPUs from guest's point of view.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><command>virsh set-user-password</command></term>
+    <listitem>
+     <para>
+      Sets the password for a user account in the guest.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><command>virsh domfsinfo</command></term>
+    <listitem>
+     <para>
+      Shows a list of mounted file systems within the running domain.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><command>virsh dompmsuspend</command></term>
+    <listitem>
+     <para>
+      Suspends a running guest.
+     </para>
+    </listitem>
+   </varlistentry>
+  </variablelist>
+ </sect1>
+ <sect1 xml:id="cha-qemu-ga-libvirt">
+  <title>Enhancing &libvirt; Commands</title>
+
+  <para>
+   If &qemu; GA is enabled inside the guest, a number of &virsh; subcommands
+   have enhanced functionality when run in the <emphasis>agent</emphasis> mode.
+   The following list includes only several examples of them. For a complete
+   list, see the &virsh; manual page and search for the
+   <literal>agent</literal> string.
+  </para>
+
+  <variablelist>
+   <varlistentry>
+    <term><command>virsh shutdown --mode agent</command> and <command>virsh reboot --mode agent</command></term>
+    <listitem>
+     <para>
+      This method of shutdown or reboot is more reliable because the shutdown
+      is performed in a clean state of the guest.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><command>virsh domfsfreeze</command> and <command>virsh domfsthaw</command></term>
+    <listitem>
+     <para>
+      Quiesces the guest file system in isolation.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><command>virsh setvcpus --guest</command></term>
+    <listitem>
+     <para>
+      Changes the number of virtual CPUs active in a guest.
```

Comment:
What does "active" mean here? This needs clarification.

I can see 2 possible explanations:

[1] There are pre-existing hi/lo limits to the number of CPUs and this moves a VM around between these limits -- e.g. that some cores/CPUs are always present, but inactive, and this activates them.

[2] That is makes the change in number of CPUs **on the fly**, i.e. while the VM is running. I think that most guest OSes will not be able to handle this, but I welcome correction on this.

Reply:
yes, it menas changing the number of vCPUs on the fly